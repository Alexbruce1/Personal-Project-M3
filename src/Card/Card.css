.card {
  background-color: rgba(242, 242, 242, .7);
  border-radius: 5px;
  box-shadow: 0 0 15px rgba(0, 0, 0, .4);
  color: #444;
  height: 35rem; 
  margin-bottom: 1rem;
  padding: 1rem;
  transition-duration: .2s;
  width: calc(100% - 2rem);
}

.card:hover {
  box-shadow: 0 0 1rem rgba(0, 0, 0, .6)
}

.card-top-container {
  display: inline-flex;
  justify-content: space-between;
  margin: 0 auto;
  width: 90%;
}

.title {
  font-weight: 900;
  /* text-decoration: underline; */
}

.title,
.media-type {
  font-family: 'Lato', sans-serif;
  font-size: 2rem;
  font-style: italic;
  max-height: 210px;
  /* overflow: hidden; */
  /* overflow-y: hidden; */
  text-align: left;
  /* white-space: nowrap; */
  /* max-width: calc(100% - 6rem); */
}

.wikipedia-link {
  background-color: rgb(242, 242, 242);
  box-shadow: 0 0 5px rgba(0, 0, 0, .4);
  border-radius: 10px;
  color: #444;
  cursor: pointer;
  font-family: 'Lato', sans-serif;
  font-size: 15px;
  font-weight: 900;
  margin: 1rem;
  padding: 8px 1rem;
  text-decoration: none;
  width: 100%;
}

.wikipedia-link:hover {
  box-shadow: 0 0 5px rgba(0, 0, 0, .6);
}

.description {
  height: 10rem;
  line-height: 165%;
  overflow: hidden;
  text-align: justify;
  text-indent: 3rem;
}

.youtube-background {
  background-color: black;
  height: 15rem;
  margin: auto;
  position: relative;
  bottom: 15rem;
  width: 90%;
}

.youtube-background {
  color: #fcfcfc;
  font-family: 'Lato', sans-serif;
  font-weight: 900;
  line-height: 15rem;  
}

.youtube-embed {
  border: none;
  box-shadow: 0 0 2rem rgba(0, 0, 0, .6);
  height: 15rem;
  margin: auto;
  padding: 0;
  position: relative;
  transition-duration: .3s;
  width: 90%;
  z-index: 1;
}

.youtube-embed:hover {
  box-shadow: 0 0 2.5rem rgba(0, 0, 0, .75);
}