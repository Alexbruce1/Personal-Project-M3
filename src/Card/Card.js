import React, { Component } from 'react';
import './Card.css';

class Card extends Component {
  render() {
    const { name, type, teaser, wiki, yt } = this.props;
    return (
      <div className="card">
        <div className="card-top-container">
          <p className="title">{name}</p>
          <p className="media-type">{ type }</p>
        </div>
        <a className='wikipedia-link' href={wiki}>Learn More</a>
        <p className="description">{ teaser }</p>
        <iframe
          className='youtube-embed' 
          title='YouTube Link' 
          src={yt}>
        </iframe>
        <div className='youtube-background'>Loading YouTube link</div>
      </div>
    )
  }
}

export default Card;