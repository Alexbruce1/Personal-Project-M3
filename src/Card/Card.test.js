import Card from './Card';
import React from 'react';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper = shallow(<Card
    name='name'
    type='type'
    teaser='teaser'
    wiki='wiki'
    yt='yt'
    yId='yId'
  />);

  it('should render like the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});