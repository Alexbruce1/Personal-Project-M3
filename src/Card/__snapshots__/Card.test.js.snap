// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render like the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    name="name"
    teaser="teaser"
    type="type"
    wiki="wiki"
    yId="yId"
    yt="yt"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-top-container"
        >
          <p
            className="title"
          >
            name
          </p>
          <p
            className="media-type"
          >
            type
          </p>
        </div>,
        <a
          className="wikipedia-link"
          href="wiki"
        >
          Learn More
        </a>,
        <p
          className="description"
        >
          teaser
        </p>,
        <iframe
          className="youtube-embed"
          src="yt"
          title="YouTube Link"
        />,
        <div
          className="youtube-background"
        >
          Loading YouTube link ...
        </div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="title"
            >
              name
            </p>,
            <p
              className="media-type"
            >
              type
            </p>,
          ],
          "className": "card-top-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "name",
              "className": "title",
            },
            "ref": null,
            "rendered": "name",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "type",
              "className": "media-type",
            },
            "ref": null,
            "rendered": "type",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Learn More",
          "className": "wikipedia-link",
          "href": "wiki",
        },
        "ref": null,
        "rendered": "Learn More",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "teaser",
          "className": "description",
        },
        "ref": null,
        "rendered": "teaser",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "youtube-embed",
          "src": "yt",
          "title": "YouTube Link",
        },
        "ref": null,
        "rendered": null,
        "type": "iframe",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading YouTube link ...",
          "className": "youtube-background",
        },
        "ref": null,
        "rendered": "Loading YouTube link ...",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-top-container"
          >
            <p
              className="title"
            >
              name
            </p>
            <p
              className="media-type"
            >
              type
            </p>
          </div>,
          <a
            className="wikipedia-link"
            href="wiki"
          >
            Learn More
          </a>,
          <p
            className="description"
          >
            teaser
          </p>,
          <iframe
            className="youtube-embed"
            src="yt"
            title="YouTube Link"
          />,
          <div
            className="youtube-background"
          >
            Loading YouTube link ...
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="title"
              >
                name
              </p>,
              <p
                className="media-type"
              >
                type
              </p>,
            ],
            "className": "card-top-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "name",
                "className": "title",
              },
              "ref": null,
              "rendered": "name",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "type",
                "className": "media-type",
              },
              "ref": null,
              "rendered": "type",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Learn More",
            "className": "wikipedia-link",
            "href": "wiki",
          },
          "ref": null,
          "rendered": "Learn More",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "teaser",
            "className": "description",
          },
          "ref": null,
          "rendered": "teaser",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "youtube-embed",
            "src": "yt",
            "title": "YouTube Link",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading YouTube link ...",
            "className": "youtube-background",
          },
          "ref": null,
          "rendered": "Loading YouTube link ...",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
